swatch-core/src/main/java/org/candlepin/subscriptions/registry/TagProfile.java

```diff
@@ -243,4 +252,45 @@ public String tagForOfferingProductName(String offeringProductName) {
   public Set<String> getOfferingProductNamesForTag(String productTag) {
     return tagToOfferingProductNamesLookup.getOrDefault(productTag, Collections.emptySet());
   }
+  // ProductRegistry methods below
+  /**
+   * Verify that the granularity requested is compatible with the finest granularity supported by
+   * the product. For example, if the requester asks for HOURLY granularity but the product only
+   * supports DAILY granularity, we can't meaningfully fulfill that request.
+   *
+   * @throws IllegalStateException if the granularities are not compatible
+   */
+  public void validateGranularityCompatibility(
+      ProductId productId, Granularity requestedGranularity) {
+    if (!tagSupportsGranularity(productId.toString(), requestedGranularity)) {
+      String msg =
+          String.format(
+              "%s does not support any granularity finer than %s",
+              productId, requestedGranularity.getValue());
+      throw new IllegalStateException(msg);
+    }
+  }
+
+  public Map<Integer, Set<String>> getEngProductIdToSwatchProductIdsMap() {
+    Map<Integer, Set<String>> engProductIdToSwatchProductIdsMap = new HashMap<>();
+    for (TagMapping tag : tagMappings) {
+      if (!"engId".equals(tag.getValueType())) continue;
+
+      Integer engId = Integer.parseInt(tag.getValue());
+
+      if (engProductIdToSwatchProductIdsMap.containsKey(engId)) {
+        throw new IllegalStateException("Duplicate engineering product ID found: " + engId);
+      }
+      engProductIdToSwatchProductIdsMap.put(engId, tag.getTags());
+    }
+    return engProductIdToSwatchProductIdsMap;
+  }
+
+  public Map<String, Set<String>> getRoleToSwatchProductIdsMap() {
+    return roleToTagLookup;
```

Comment:
This method can removed and the `@Getter` annotation can be used on the field. Probably best if we rename the field to match the getter method that you are removing so that it makes more sense when accessed.